homeassistant/components/broadlink/updater.py

```diff
@@ -29,6 +38,68 @@ def get_update_manager(device):
     return update_managers[device.api.type](device)
 
 
+class BroadlinkScout:
+    """Manages device discovery."""
+
+    def __init__(self, hass):
+        """Initialize the scout."""
+        self.hass = hass
+        self.timeout = 5
+        self.coordinator = DataUpdateCoordinator(
+            hass,
+            _LOGGER,
+            name="discovery",
+            update_method=self.async_discover,
+            update_interval=timedelta(minutes=15),
+            request_refresh_debouncer=debounce.Debouncer(
+                hass, _LOGGER, cooldown=self.timeout, immediate=True
+            ),
+        )
+        self.coordinator.async_add_listener(self.update_listener)
```

Comment:
I'm not sure I'm not committing a crime. Am I allowed to subscribe to updates here?

Reply:
I think it should be part of `async def async_added_to_hass(self):` if I look at others (including our integration)?

Reply:
This is not an entity, we are inside the component, so we don't have this method available.

Reply:
I stand corrected than, your question remains open

Reply:
I could implement a destructor by creating an exclusive config entry for the integration discovery. But this idea sounds strange to me because we don't have a singleton for the integration. Each device has its own flow and entry.

The initial plan was to deal with device discovery at the component level, but the component does not have an unload() method, so I didn't find an official way of killing this listener.

@MartinHjelmare May I have your opinion on this?

Reply:
I'd move this to `async_setup_entry` and only do it if it doesn't exist in hass.data already.

In `async_unload_entry` check if it's the last config entry and then unregister the listener and remove the coordinator from hass.data.

Reply:
I will do it. Thank you!